### PR TITLE
Two way for #89

### PR DESCRIPTION
**Pull request by saighost**
#89 
made partial and interface to apply recursively
ignore the object key with an undefined value and without a default value.

### CODE REVIEW

**Comment by ianstormtaylor** [comment]
Hey @saighost can you turn off the formatting stuff? I think I like the direction of this but it's hard to tell with the formatting noise in the PR. Thanks!

**Comment by saighost** [comment]
sorry, I will fix it,

File: src/kinds.js [hunk 1]
```diff
@@ -285,7 +285,12 @@ function inter(schema, defaults, options) {
   for (const key in schema) {
     ks.push(key)
     const s = schema[key]
-    const kind = any(s, undefined, options)
+    let kind
+    if (kindOf(s) === 'object') {
+      kind = inter(s, undefined, options)
+    } else {
+      kind = any(s, undefined, options)
+    }
```
Comment (ianstormtaylor) [inline src/kinds.js]: Hey @saighost would you mind changing these lines to be ternaries, like:

```
const kind = kindOf(s) === 'object' ? ...
```

File: src/kinds.js [hunk 2]
```diff
@@ -552,7 +559,12 @@ function partial(schema, defaults, options) {
   for (const key in schema) {
     ks.push(key)
     const s = schema[key]
-    const kind = any(s, undefined, options)
+    let kind
+    if (kindOf(s) === 'object') {
+      kind = partial(s, undefined, options)
+    } else {
+      kind = any(s, undefined, options)
+    }
```
Comment (ianstormtaylor) [inline src/kinds.js]: Hey @saighost would you mind changing these lines to be ternaries, like:

```
const kind = kindOf(s) === 'object' ? ...
```

File: src/kinds.js [hunk 3]
```diff
@@ -483,8 +488,10 @@ function object(schema, defaults, options) {
         const d = defaults && defaults[key]
         v = resolveDefaults(d, value)
       }
-
       if (!kind) {
+        if (v === undefined) {
+          return
+        }
```
Comment (ianstormtaylor) [inline src/kinds.js]: What's this extra check needed for?

Comment (saighost) [reply src/kinds.js]: oh, for a key with an undefined value.
My data is wrapped by Proxy to filter and format value, and when an unknown key is appeared, I can only return an undefined value.

Comment (saighost) [reply src/kinds.js]: I try partial, but it can't ignore children's properties.

Comment (ianstormtaylor) [reply src/kinds.js]: Could you open a separate issue/PR for this so we can discuss it? It seems unrelated to the partial/object issue.

File: src/kinds.js (hunk 1 above)
Comment (saighost) [reply src/kinds.js]: Finish.

File: src/kinds.js (hunk 3 above)
Comment (saighost) [reply src/kinds.js]: #89 has a small test case.
I can try to add a test case for it.

**Comment by ianstormtaylor** [comment]
Hey @saighost, thank you for this PR. I believe that as of #139 with the changes to `struct.interface` and `struct.partial` this makes less sense now, so I'm going to close it. But if you disagree, feel free to reply!

**Pull request closed without merging**